frontend/src/components/product/components/ProductWrapper.tsx

```diff
@@ -12,7 +12,7 @@ const ProductWrapper: FC<ProductProps> = ({ product }: ProductProps) => {
   return (
     <div>
       <p className="text-lg">
-        ค้นพบน้ำหอมทั้งหมด <span className="font-semibold">{product.length}</span> รายการ
+        ค้นพบน้ำหอมทั้งหมด <span className="font-semibold">{product?.length}</span> รายการ
```

Comment:
This can be `undefined`.
```suggestion
        ค้นพบน้ำหอมทั้งหมด <span className="font-semibold">{product?.length && "0"}</span> รายการ
```